### DISCUSSION TITLE
Example RestAPI using FastAPI and pyodbc to connect to a MSSQL Database

### DISCUSSION THREAD

**Discussion by esundberg** [fastapi/fastapi#7606] [Questions]
Created a sample RestAPI using FastAPI and pyodbc to connect to a MSSQL Database. Good for learning and seeing a complete example that is a 3rd party library to connect to a database.

FastAPI Examples
- POST, GET (One and All Records), PUT, and DELETE
- How to call a CRUD Class to interface with a database
- Request Body parsing with pydantic for POST's and PUT's
- How to get the raw request body using scarlette request.

Python Class to interface to the database
- Create record
- List all records
- Read one record
- Update one record
- Delete one record

[https://github.com/esundberg/FastAPI-pyodbc-mssql-example](https://github.com/esundberg/FastAPI-pyodbc-mssql-example)

Hope this could help someone.

**Comment by esundberg** [comment]
Closing

**Comment by tiangolo** [comment]
I assume you solved your problem, thanks for closing the issue :+1:

**Comment by szrg** [comment]
Bad example. pyodbc is not async-aware, so every database call is synchronous and will block the server. You should rather use aioodbc.
